src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,416 @@
+import { MediaUIAttributes } from './constants.js';
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import { Document as document } from './utils/server-safe-globals.js';
+
+const DEFAULT_LISTBOX_LABEL = 'listbox';
+
+const template = document.createElement('template');
+template.innerHTML = `
+  <style>
+    [role="listbox"] {
+        margin: 1em 0 0;
+        padding: 0;
+        min-height: 18em;
+        border: 1px solid #aaa;
+        background: white;
+    }
+    
+    [role="listbox"]#listbox {
+        max-height: 18em;
+        overflow-y: auto;
+    }
+    
+    [role="listbox"] + *,
+    .listbox-label + * {
+        margin-top: 1em;
+    }
+    
+    [role="option"] {
+        display: block;
+        padding: 0 1em 0 1.5em;
+        line-height: 1.8em;
+    }
+    
+    :host(:focus) [role="option"].focused,
+    :host(:focus-within) [role="option"].focused {
+        background: #bde4ff;
+    }
+    
+    [role="option"][aria-selected="true"]::before {
+        left: 0.5em;
+        content: "✓";
+    }
+  </style>
+  <div>
+    <span id="listbox-label" tabindex="-1" class="listbox-label">${DEFAULT_LISTBOX_LABEL}</span>
+    <ul id="listbox" role="listbox" aria-labelledby="listbox-label">
+    </ul>
+  </div>
+`;
+
+const isNil = (x) => x == undefined;
+const isIterable = (value) => {
+  return Symbol.iterator in Object(value);
+};
+const isArraylike = (value) => {
+  return (
+    typeof value?.length === 'number' &&
+    value.length >= 0 &&
+    value.length <= 4294967295
+  );
+};
+const identity = (x) => x;
+
+const getClassName = (x) => {
+  return x?.constructor?.name ?? 'UNKNOWN';
+};
+
+const DEFAULT_FORMAT = identity;
+const DEFAULT_ITEM_RENDERER = 'li';
```

Comment:
currently defining an "itemRenderer" as an element (which could be a custom element) that is expected to have innerText + attributes set on it (See, e.g., `render()` below)